indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamDataSourceMetadata.java

```diff
@@ -59,6 +59,18 @@ public boolean matches(DataSourceMetadata other)
     return plus(other).equals(other.plus(this));
   }
 
+  @Override
+  public int compareTo(DataSourceMetadata other)
+  {
+    if (!getClass().equals(other.getClass())) {
+      return -1;
+    }
```

Comment:
this seems inconsistent with other implementations where we throw an exception when we encounter a different class.